geniza/annotations/views.py

```diff
@@ -74,11 +76,24 @@ def parse_annotation_data(request):
         )
     except Footnote.DoesNotExist:
         source = Source.objects.get(pk=source_id)
+        # try to find a corresponding non-digital footnote
+        # (i.e. Translation for Digital Translation, Edition for Digital Edition)
+        corresponding_footnote = Footnote.objects.filter(
+            doc_relation__contains=corresponding_relation,
+            source__pk=source_id,
+            content_type=document_contenttype,
+            object_id=document_id,
```

Comment:
you might limit your query to footnotes that actually have a location set; from what I remember a lot of them do not, and handling at the query level might simplify the logic below

Reply:
oh, and I see you want one and only one result, otherwise it isn't useful...  I don't know if filtering on location makes that problematic, but another way to make sure you get one is to call `get` instead of filter and catch the too many results exception that Django will throw; might simplify the logic where you're setting the location below

Reply:
Great ideas, thank you @rlskoeser!